### PR TITLE
Better error message when no wiki

### LINKED ISSUE #27: Add more informative error message when wiki doesn't exist
image

I was scratching my head for a few minutes before I remembered that I needed to create a dummy wiki page to init it. 🙄 If I forgot that, others will too. Having a nice error message is undoubtedly a good idea 🎉

### PR DESCRIPTION
**Pull request by jcbhmr**
⚠ this pr is dependent on #35

This PR would...
- [x] Create a git_exists.sh file
- [x] use source to import git_exists.sh
- [x] use the bash lib pattern to export a git_exists function
- [x] properly use absolute paths to `source $abs_path` instead of `source $rel_path`
- [x] make the error when the wiki repo doesn't exist more descriptive of what went wrong
- [x] make the error when the wiki repo doesn't exist give a suggestion
- [x] make the error when the wiki repo doesn't exist point to documentation
- [x] use common cli tool logging conventions (see below)

Common CLI logging convention is to report some kind of useful token to stdout. in this case, that's the same thing as the input after confirming that it exists.

```sh
$ src/git_exists.sh github.com/spenserblack/actions-wiki
github.com/spenserblack/actions-wiki
```

it's also common to prefix errors with the cli tool where they originated from. in this case, we use `$(basename "$0")` as a prefix. this means that an imported script will if `git_exists()` has an error, it will say `index.sh` has an error. This could be changed to use `$(basename "${BASH_SOURCE[0]}")` instead if desired

```sh
$ src/git_exists.sh nope
git_exists.sh: nope is not a Git repository
$ ./thing_that_uses_git_exists.sh nope
thing_that_uses_git_exists.sh: nope is not a Git repository
```

vs

```sh
$ src/git_exists.sh nope
git_exists.sh: nope is not a Git repository
$ ./thing_that_uses_git_exists.sh nope
git_exists.sh: nope is not a Git repository
```

closes #27

### CODE REVIEW

**Comment by jcbhmr** [comment]
@spenserblack requesting review (i can't do it in the gh ui for some reason)
image

File: src/index.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+set -e
+
+source "$(dirname "${BASH_SOURCE[0]}")/git_exists.sh"
+
+if ! git_exists "https://$INPUTS_TOKEN@$INPUTS_REPOSITORY.wiki.git" &>/dev/null; then
+  echo "$(basename "$0"): https://$INPUTS_REPOSITORY.wiki.git doesn't exist." >&2
+  echo 'Did you remember to create the first wiki page manually?' >&2
+  echo 'You can find more information on the readme. [1]' >&2
+  echo
+  echo '[1]: https://github.com/spenserblack/actions-wiki#usage'
```
Comment (spenserblack) [inline src/index.sh:11]: This makes me wonder if there's a built-in context or environment variable to get the URL of the action being used.

**Comment by spenserblack** [comment]
:thinking: I swear I didn't have the button to convert others to draft, but I was able to do it with this one.

Converting to draft so I don't forget and merge this before #35

**Review by spenserblack — APPROVED** [review]
LGTM!

Want to resolve the merge conflicts?

Just a warning, I'll probably be squash-merging often.

File: src/index.sh
Comment (spenserblack) [inline src/index.sh:1]: File mode changed back in this PR

File: src/git_exists.sh
Comment (spenserblack) [inline src/git_exists.sh:1]: Maybe also make this executable?

BTW, if it's more convenient for you, there's `git add --chmod=+x ...`

File: src/index.sh [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
 git config --local user.name github-actions
 git config --local user.email github-actions@github.com
-git remote add origin https://$INPUT_TOKEN@$INPUT_REPOSITORY.wiki.git
+git remote add origin https://$INPUTS_TOKEN@$INPUTS_REPOSITORY.wiki.git
 git fetch origin
 git reset origin/master
 
 git add --all
-git commit -m "$INPUT_COMMIT_MESSAGE"
+git commit -m "$INPUTS_COMMIT_MESSAGE"
```
Comment (spenserblack) [inline src/index.sh:25]: These should be `INPUT_...`, right?

**Review by spenserblack — CHANGES_REQUESTED** [review]

File: src/index.sh
Comment (jcbhmr) [reply src/index.sh:1]: oop 🤦

File: src/index.sh (hunk 2 above)
Comment (jcbhmr) [reply src/index.sh]: fixed in https://github.com/spenserblack/actions-wiki/pull/37/commits/a5b0b949fec5746d2e9a6e12c2b2da49fb73bb2d

File: src/git_exists.sh
Comment (jcbhmr) [reply src/git_exists.sh:1]: i made it explicitly non-exe-like

File: src/index.sh
Comment (jcbhmr) [reply src/index.sh:1]: fixed in https://github.com/spenserblack/actions-wiki/pull/37/commits/1ca6b7ec9dc9655e7534c75015c61c3e2870b84c

**Comment by jcbhmr** [comment]
@spenserblack ok this time i _actually_ fixed merge conflicts -- i hope 🙏

File: src/index.sh [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 git config --local user.name github-actions
 git config --local user.email github-actions@github.com
-git remote add origin https://$INPUT_TOKEN@$INPUT_REPOSITORY.wiki.git
+git remote add origin "https://$INPUT_TOKEN@$INPUT_REPOSITORY.wiki.git"
 git fetch origin
 git reset origin/master
 
 git add --all
-git commit -m "$INPUT_COMMIT_MESSAGE"
+git commit -m "$INPUTS_COMMIT_MESSAGE"
```
Comment (spenserblack) [inline src/index.sh:26]:
```suggestion
git commit -m "$INPUT_COMMIT_MESSAGE"
```

**Review by spenserblack — APPROVED** [review]

File: src/index.sh (hunk 3 above)
Comment (jcbhmr) [reply src/index.sh]: oh i guess i missed one oops ❤

**Comment by jcbhmr** [comment]
@spenserblack #38 is merged now

**Comment by jcbhmr** [comment]
todo(@jcbhmr): rebase against new main branch

**Comment by jcbhmr** [comment]
@spenserblack ready2merge

**Pull request merged**